Assets/LeapMotion/Scripts/LeapImageRetriever.cs

```diff
@@ -322,41 +321,42 @@ public class EyeTextureData {
     }
   }
 
-  
   void OnPreRender() {
-    Controller controller = provider.GetLeapController();
-    long start = controller.Now();
-    if (!_requestedImage.IsComplete) _lateImages++;
-    while (!_requestedImage.IsComplete) {
-      if (controller.Now() - start > 9000) break;
-    }
-    if (_requestedImage.IsComplete) {
-      //Debug.Log("controller_.Now():" + controller.Now() + " - CurrentFrame.Timestamp:" + image.Timestamp + " = " + (controller.Now() - image.Timestamp));
-      //Debug.Log("LeapImageRetriever.OnPreRender image.SequenceId: " + image.SequenceId);
-      if (_eyeTextureData.CheckStale(_requestedImage, _requestedImage)) {
-        _eyeTextureData.Reconstruct(_requestedImage, _requestedImage);
+    if(ImagesEnabled){
+      Controller controller = provider.GetLeapController();
+      long start = controller.Now();
+      while (!_requestedImage.IsComplete) {
+        if (controller.Now() - start > ImageTimeout) break;
+      }
+      if (_requestedImage.IsComplete) {
+        if (_eyeTextureData.CheckStale(_requestedImage, _requestedImage)) {
+          _eyeTextureData.Reconstruct(_requestedImage, _requestedImage);
+        }
+        _eyeTextureData.UpdateTextures(_requestedImage, _requestedImage);
+      } else if(!checkingImageState){
+        StartCoroutine(CheckImageMode(2));
+        checkingImageState = true;
```

Comment:
You can put this set to true inside of CheckImageMode (before the waitforseconds), that way starting it is always a 1 line operation.